pkg/modelzooserver/modelzooserver.go

```diff
@@ -321,8 +322,125 @@ func (s *modelZooServer) DropModelRepo(ctx context.Context, req *pb.ReleaseModel
 	return &pb.ReleaseResponse{Success: true, Message: ""}, nil
 }
 
-func (s *modelZooServer) ReleaseModel(stream pb.ModelZooServer_ReleaseModelServer) error {
-	var req *pb.ReleaseModelRequest
+func (s *modelZooServer) ReleaseModel(ctx context.Context, req *pb.ReleaseModelRequest) (*pb.ReleaseResponse, error) {
+	if err := checkNameAndTag(req.GetName(), req.GetTag()); err != nil {
+		return nil, err
+	}
+	// Create database sqlflow_public_models to store public trained models.
+	if _, err := s.DB.Exec(fmt.Sprintf("CREATE DATABASE IF NOT EXISTS %s;", publicModelDB)); err != nil {
+		return nil, err
+	}
+	// download model from user's model database storage
+	db, err := database.OpenDB(req.DbConnStr)
+	if err != nil {
+		return nil, err
+	}
+	defer db.Close()
+	dir, err := ioutil.TempDir("/tmp", "upload_model_zoo")
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(dir)
+	tarFile, err := model.DumpDBModel(db, req.Name, dir)
+	if err != nil {
+		return nil, err
+	}
+	modelMeta, err := model.ExtractMetaFromTarball(tarFile, dir)
+	if err != nil {
+		return nil, err
+	}
+	sendFile, err := os.Open(tarFile)
+	if err != nil {
+		return nil, err
+	}
+	defer sendFile.Close()
+
+	// model name is the original INTO clause, it can be like db.table
+	// reformat it to db_table as the table name in model zoo database.
+	modelTableName := fmt.Sprintf("%s.%s", publicModelDB, strings.ReplaceAll(req.Name, ".", "_"))
```

Comment:
This is not compatible with local release mode, local mode only use the table name. We can make them the same.